Source/CarthageKit/Project.swift

```diff
@@ -811,9 +831,14 @@ public final class Project { // swiftlint:disable:this type_body_length
 			}
 
 		return loadResolvedCartfile()
-			.map { resolvedCartfile -> [(Dependency, PinnedVersion)] in
+			.flatMap(.latest) { resolvedCartfile in
+				return self.transitiveDependencies(dependenciesToCheckout, resolvedCartfile: resolvedCartfile)
+			}
+			.zip(with: loadResolvedCartfile())
+			.map { subDependencies, resolvedCartfile -> [(Dependency, PinnedVersion)] in
+				let dependencies = (dependenciesToCheckout ?? []) + subDependencies
 				return resolvedCartfile.dependencies
-					.filter { dep, _ in dependenciesToCheckout?.contains(dep.name) ?? true }
+					.filter { dep, _ in dependencies.isEmpty ? true : dependencies.contains(dep.name) }
```

Comment:
I think we should keep this line the same, where `nil` means everything.

Probably the best way to do that would be to add these inside the `flatMap`.

```swift
.flatMap(.latest) { resolvedCartfile in
    guard let dependenciesToCheckout = dependenciesToCheckout else {
        return SignalProducer(value: (nil, resolvedCartfile))
    }
    return self
        .transitiveDependencies(dependenciesToCheckout, resolvedCartfile: resolvedCartfile)
        .map { (dependenciesToCheckout + $0, resolvedCartfile) }
}
.map { dependenciesToCheckout, resolvedCartfile in
    return resolvedCartfile.dependencies
        .filter { dep, _ in dependenciesToCheckout?.contains(dep.name) ?? true }
}